app/post.go

```diff
@@ -2082,6 +2082,68 @@ func (a *App) CheckPostReminders() {
 
 }
 
+func (a *App) GetPostInfo(c request.CTX, postID string) (*model.PostInfo, *model.AppError) {
+	userID := c.Session().UserId
+	post, appErr := a.GetSinglePost(postID, false)
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	channel, appErr := a.GetChannel(c, post.ChannelId)
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	notFoundError := model.NewAppError("GetPostInfo", "app.post.get.app_error", nil, "", http.StatusNotFound)
+
+	var team *model.Team
+	hasPermissionToAccessTeam := false
+	if channel.TeamId != "" {
+		team, appErr = a.GetTeam(channel.TeamId)
+		if appErr != nil {
+			return nil, appErr
+		}
+
+		if team.Type == model.TeamOpen {
+			hasPermissionToAccessTeam = a.HasPermissionToTeam(userID, team.Id, model.PermissionJoinPublicTeams)
+		} else if team.Type == model.TeamInvite {
+			hasPermissionToAccessTeam = a.HasPermissionToTeam(userID, team.Id, model.PermissionJoinPrivateTeams)
```

Comment:
@mkraft @kostaspt It's my understanding that sysadmins can join teams that are marked as invite only (private) so when we use `HasPermissionToTeam` it will return the team type as open because technically a sysadmin can join an invite only team. The problem is that I think this ticket wants the response to tell a sysadmin that a team is technically invite only (private) regardless of their sysadmin role? Right now the response will return the team type as open when in fact it should be private.

Example response from a private team that a sysadmin is not part of
```
{
    "channel_id": "j7oz3aqeztfsjcs3sgozb68mxy",
    "channel_type": "P",
    "channel_display_name": "Private",
    "team_id": "ifr6orf37tdwmrk5tyzd3mfehy",
    "team_type": "O",
    "team_display_name": "test"
}
```

`Whether the team is private will be use to alert when a sysadmin is trying to join a private team.`

cc/ @larkox for visibility since you created the original ticket

Reply:
Yes, I thought the same when I first read that part of the description. But for this use case, there is probably already knowledge of that user being a sysadmin, so that the messages can be created based on that. But yes, maybe there is a need for better clarification so we can do it correctly the first time.

Reply:
It is important to know the type of the team. In terms of notification, it has not the same importance joining an open team than a private team.
The only ones that can join private teams are sysadmins, so it is the particular case where it is important to have a difference. Any other user will probably only be able to join open teams.

Reply:
@larkox Sorry for the delayed response. From your answer, I understand that we'll need the type of team/channel (which is already implemented) and some way to differentiate the type of the user so the "user type" or at least a boolean if the user is sysadmin (which I'll add).

Another thing that sounds useful to me is maybe to have an `is_channel_member:bool` and `is_team_member:bool` so we can avoid checking if there is even a need to try to join to get to the post.

Reply:
The type of the user is not so important (I know what kind of user I am). It is about the type of team. Probably I can get the team to check it, but that would be another api call, so would be better if this call already give me that information.

Regarding the is_member, it could be checked locally (in the redux / watermelondb information) but if it comes "for free" getting it from the server, yes, that sounds great.